### ISSUE TITLE
strict mode: missing type "object" for every "if" inside "allOf"

### ISSUE THREAD

**Issue by mummybot** [ajv-validator/ajv#1606]
**What version of Ajv are you using? Does the issue happen if you use the latest version?**
v8.1.0

**Ajv options object**
Any combination of

```javascript
strictTypes: false,
strict: false,
strictSchema: false,
```

**JSON Schema**

```json
"allOf": [
  "if": {
    "properties": {}
  },
  "then": {}
]
```

**Validation result, data AFTER validation, error messages**
```
strict mode: missing type "object" for keyword "properties" at "#" (strictTypes)
```

**What results did you expect?**
No warnings

**Are you going to resolve the issue?**
Adding `"type": "object," prevents the warnings, but is a lot of 'decoration' for schema which otherwise are valid.

```javascript
"allOf": [
  "if": {
    "type": "object",
    "properties": {}
  },
  "then": {}
]
```

**Comment by mummybot** [comment]
Nevermind, this isn't an issue!

We are running Ajv twice(!) in our code base on the same schema (don't ask), and the second instance was throwing the warnings. Adding the correct options there removed the warnings.

**Comment by epoberezkin** [comment]
Adding type: object to schema root should also remove this warning - it’s likely you don’t have type defined anywhere - was it the case?

**Comment by mummybot** [comment]
To be more specific: we have created a configurable form builder which uses JSON schema to both drive what fields are displayed, and what validation occurs. We have our master Ajv instance validating the entire schema, and then a further instance checking the validation status at an `if` node to determine if the UI displays a `then` or an `else`. I was missing adding `strictTypes: false` on this second instance.
